File: lingting-spring-web/src/main/java/live/lingting/spring/web/converter/StringToLocalTimeConverter.java
package live.lingting.spring.web.converter;

import live.lingting.framework.util.LocalDateTimeUtils;
import org.springframework.core.convert.TypeDescriptor;
import org.springframework.stereotype.Component;

import java.time.LocalTime;
import java.util.Collections;
import java.util.Set;

/**
 * @author lingting 2022/9/28 12:17
 */
@Component
public class StringToLocalTimeConverter implements Converter<LocalTime> {

	@Override
	public boolean matches(TypeDescriptor sourceType, TypeDescriptor targetType) {
		return String.class.isAssignableFrom(sourceType.getType())
				&& LocalTime.class.isAssignableFrom(targetType.getType());
	}

	@Override
	public Set<ConvertiblePair> getConvertibleTypes() {
		return Collections.singleton(new ConvertiblePair(String.class, LocalTime.class));
	}

	@Override
	public LocalTime convert(Object source, TypeDescriptor sourceType, TypeDescriptor targetType) {
		String string = (String) source;
		return LocalDateTimeUtils.parseTime(string);
	}

}

File: lingting-spring-web/src/main/kotlin/live/lingting/spring/web/converter/StringToLocalTimeConverter.kt
package live.lingting.spring.web.converter

import java.time.LocalTime
import live.lingting.framework.util.LocalTimeUtils
import org.springframework.core.convert.TypeDescriptor
import org.springframework.core.convert.converter.GenericConverter.ConvertiblePair

/**
 * @author lingting 2022/9/28 12:17
 */
class StringToLocalTimeConverter : Converter<LocalTime> {
    override fun matches(sourceType: TypeDescriptor, targetType: TypeDescriptor): Boolean {
        return String::class.java.isAssignableFrom(sourceType.type)
                && LocalTime::class.java.isAssignableFrom(targetType.type)
    }

    override fun getConvertibleTypes(): MutableSet<ConvertiblePair> {
        return mutableSetOf<ConvertiblePair>(ConvertiblePair(String::class.java, LocalTime::class.java))
    }

    override fun convert(source: Any?, sourceType: TypeDescriptor, targetType: TypeDescriptor): LocalTime {
        val string = source as String
        return LocalTimeUtils.parse(string)
    }
}
